tests/async/AsyncWatchmanTestCase.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2012-present Facebook, Inc.
+# Licensed under the Apache License, Version 2.0
+
+import errno
+import unittest
+import time
+import tempfile
+import os.path
+import os
+import WatchmanInstance
+import copy
+import sys
+import glob
+import asyncio
+
+from pywatchman.aioclient import AIOClient as WatchmanClient
+from pywatchman import encoding
+
+import pywatchman
+
+def norm_path(name):
+    return os.path.normcase(os.path.normpath(name))
+
+# TODO: This normalization will not be needed once we have full unicode support
+# in place as per
+# https://github.com/facebook/watchman/wiki/Better-Unicode-handling-plan
+def conv_path_to_bytes(path):
+    if isinstance(path, str):
+        return path.encode('utf8')
+    else:
+        return path
+
+class AsyncWatchmanTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.loop = asyncio.get_event_loop()
+        self.client = self.loop.run_until_complete(WatchmanClient.from_socket())
+
+    def tearDown(self):
+        self.client.close()
+
+    def run(self, result):
+        if result is None:
+            raise Exception('MUST be a runtests.py:Result instance')
+        super(AsyncWatchmanTestCase, self).run(result)
+        return result
+
+    def mkdtemp(self):
```

Comment:
What's the point of this method? You're importing `tempfile` in the test case file anyway.